src/recordpage.cpp

```diff
@@ -834,6 +856,46 @@ void RecordPage::setError(const QString& message)
     ui->applicationRecordErrorMessage->setVisible(!message.isEmpty());
 }
 
+void RecordPage::addKonsoleWidget()
+{
+    KService::Ptr service = KService::serviceByDesktopName(QStringLiteral("konsolepart"));
+
+    if (!service) {
+        return;
+    }
+
+    const auto tail = QStandardPaths::findExecutable(QStringLiteral("tail"));
+
+    if (tail.isEmpty()) {
+        return;
+    }
+
+    if (m_konsolePart) {
+        ui->recordOutputBox->layout()->removeWidget(m_konsolePart->widget());
+        m_konsolePart->deleteLater();
+    }
+    m_konsolePart = service->createInstance<KParts::ReadOnlyPart>(this);
+
+    if (!m_konsolePart) {
+        return;
+    }
+
+    m_konsoleFile = new QTemporaryFile(this);
```

Comment:
move down, below the cast check of the terminalInterface

also check if m_konsoleFile exists already and if so, delete it before recreating it here. otherwise we are "leaking" temporary files until hotspot is quit at the very end